Port tests to python

1000101 opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
#72828

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

Commit:
nixos/fancontrol: port test to python

Commit:
nixos/deluge: port test to python

Commit:
nixos/clickhouse: port test to python

Commit:
nixos/firewall: port test to python

Commit:
nixos/fish: port test to python

Commit:
nixos/fluentd: port test to python

Commit:
nixos/hibernate: port test to python

Commit:
nixos/hound: port test to python

Commit:
nixos/handbrake: port test to python

Commit:
nixos/icingaweb2: port test to python

Commit:
nixos/incron: port test to python

Commit:
nixos/minidlna: port test to python

Commit:
nixos/jackett: port test to python

Commit:
nixos/miniflux: port test to python

mmahut commented:
@GrahamcOfBorg test roundcube
@GrahamcOfBorg test fancontrol
@GrahamcOfBorg test deluge
@GrahamcOfBorg test clickhouse
@GrahamcOfBorg test firewall
@GrahamcOfBorg test fish
@GrahamcOfBorg test fluentd
@GrahamcOfBorg test hibernate
@GrahamcOfBorg test hound
@GrahamcOfBorg test handbrake
@GrahamcOfBorg test icingaweb2
@GrahamcOfBorg test incron
@GrahamcOfBorg test minidlna
@GrahamcOfBorg test jackett
@GrahamcOfBorg test miniflux
@GrahamcOfBorg test haproxy

mmahut commented:
@GrahamcOfBorg test roundcube
@GrahamcOfBorg test fancontrol
@GrahamcOfBorg test deluge
@GrahamcOfBorg test clickhouse
@GrahamcOfBorg test firewall
@GrahamcOfBorg test fish
@GrahamcOfBorg test fluentd
@GrahamcOfBorg test hibernate
@GrahamcOfBorg test hound
@GrahamcOfBorg test handbrake
@GrahamcOfBorg test icingaweb2
@GrahamcOfBorg test incron
@GrahamcOfBorg test minidlna
@GrahamcOfBorg test jackett
@GrahamcOfBorg test miniflux
@GrahamcOfBorg test haproxy

blitz commented:
This looks like a pretty straight forward and mechanical conversion. I spotted no issues.
@tfc Ping.

tfc commented on nixos/tests/roundcube.nix:
```diff
@@ -21,7 +21,7 @@ import ./make-test-python.nix ({ pkgs, ...} : {
 
   testScript = ''
-    roundcube.start
+    roundcube.start()
```
oh good catch... :+1:

1000101 commented:
Should I drop the tests that are failing on arm or is this good to go?

flokli commented:
@1000101 did these fail on arm previously, too?

Can you rebase on latest master?

1000101 commented:
> @1000101 did these fail on arm previously, too?
> 
> Can you rebase on latest master?

Jackett has been failing before:
jackett: https://github.com/NixOS/nixpkgs/pull/72529

Not sure about the hibernate though.

Rebased!

mmahut commented:
@GrahamcOfBorg test roundcube
@GrahamcOfBorg test fancontrol
@GrahamcOfBorg test deluge
@GrahamcOfBorg test clickhouse
@GrahamcOfBorg test firewall
@GrahamcOfBorg test fish
@GrahamcOfBorg test fluentd
@GrahamcOfBorg test hibernate
@GrahamcOfBorg test hound
@GrahamcOfBorg test handbrake
@GrahamcOfBorg test icingaweb2
@GrahamcOfBorg test incron
@GrahamcOfBorg test minidlna
@GrahamcOfBorg test jackett
@GrahamcOfBorg test miniflux
@GrahamcOfBorg test haproxy

flokli commented:
This is a bit hard to merge, can this be broken down to individual PRs?

flokli commented:
Ok, it was mostly the roundcube fix already fixed in e292be76fc79dfe1ff1e2680a00c33864caf6a9e. Dropped that commit and rebased on master.

flokli commented:
Built and succeeded running all updated tests locally (x86_64), merging.

1000101 commented:
> Ok, it was mostly the roundcube fix already fixed in [e292be7](https://github.com/NixOS/nixpkgs/commit/e292be76fc79dfe1ff1e2680a00c33864caf6a9e). Dropped that commit and rebased on master.

Thanks a million!!!
